services/dynamic-scheduler/src/simcore_service_dynamic_scheduler/api/rest/_health.py

```diff
@@ -1,10 +1,40 @@
+from typing import Annotated
+
 import arrow
-from fastapi import APIRouter
+from fastapi import APIRouter, Depends
 from fastapi.responses import PlainTextResponse
+from models_library.errors import (
+    RABBITMQ_CLIENT_UNHEALTHY_MSG,
+    REDIS_CLIENT_UNHEALTHY_MSG,
+)
+from servicelib.rabbitmq import RabbitMQClient, RabbitMQRPCClient
+from servicelib.redis import RedisClientSDK
+
+from ._dependencies import (
+    get_rabbitmq_client_from_request,
+    get_rabbitmq_rpc_server_from_request,
+    get_redis_client_from_request,
+)
 
 router = APIRouter()
 
 
+class HealthCheckError(RuntimeError):
+    """Failed a health check"""
+
+
 @router.get("/", response_class=PlainTextResponse)
-async def healthcheck():
+async def healthcheck(
+    rabbit_client: Annotated[RabbitMQClient, Depends(get_rabbitmq_client_from_request)],
+    rabbit_rpc_server: Annotated[
+        RabbitMQRPCClient, Depends(get_rabbitmq_rpc_server_from_request)
+    ],
+    redis_client_sdk: Annotated[RedisClientSDK, Depends(get_redis_client_from_request)],
+):
+    if not rabbit_client.healthy or not rabbit_rpc_server.healthy:
+        raise HealthCheckError(RABBITMQ_CLIENT_UNHEALTHY_MSG)
```

Comment:
Q: what is the purpose of adding messages? AFAIK our [healthcheck script](https://github.com/itisfoundation/osparc-simcore/blob/8a42137702a79fdf3520c755c66530122241462b/services/web/server/docker/healthcheck.py#L30-L40) (the one run by the daemon) only checks whether it was 200 or not .  Unfortunately our script is very poor and eventually we only see this info from the docker inspect

<img width="763" alt="image" src="https://github.com/ITISFoundation/osparc-simcore/assets/32402063/eca3498c-5ced-4ff5-83b8-c7f2e178582e">

Therefore, I am not saying it is a bad idea but IMO we overdo the healthcheck since later we just do not grab all the info provided in them